src/libmeasurement_kit/net/ssl_context_impl.hpp

```diff
@@ -34,26 +35,41 @@ static void initialize_ssl() {
     }
 }
 
-template <MK_MOCK(SSL_CTX_new), MK_MOCK(SSL_CTX_load_verify_locations)>
+template <MK_MOCK(SSL_CTX_new), MK_MOCK(SSL_CTX_load_verify_locations)
+#ifdef LIBRESSL_VERSION_NUMBER
+          , MK_MOCK(SSL_CTX_load_verify_mem)
+#endif
+          >
 ErrorOr<SSL_CTX *> make_ssl_ctx(std::string path) {
 
     debug("ssl: creating ssl context with bundle %s", path.c_str());
     initialize_ssl();
 
-    if (path == "") {
-        return MissingCaBundlePathError();
-    }
-
     SSL_CTX *ctx = SSL_CTX_new(TLSv1_client_method());
     if (ctx == nullptr) {
         debug("ssl: failed to create SSL_CTX");
         return SslCtxNewError();
     }
 
-    if (SSL_CTX_load_verify_locations(ctx, path.c_str(), nullptr) != 1) {
-        debug("ssl: failed to load verify location");
+    if (path != "") {
+        if (!SSL_CTX_load_verify_locations(ctx, path.c_str(), nullptr)) {
+            debug("ssl: failed to load verify location");
+            SSL_CTX_free(ctx);
+            return SslCtxLoadVerifyLocationsError();
+        }
+    } else {
+#ifdef LIBRESSL_VERSION_NUMBER
+        std::vector<uint8_t> bundle = builtin_ca_bundle();
+        if (!SSL_CTX_load_verify_mem(ctx, bundle.data(), bundle.size())) {
```

Comment:
Good catch, thanks! I have implemented a conditional check such that we don't even attempt to use that function if we're under libressl < 2.1.4 (see ce5bd8e).